modules/apps/27-interchain-accounts/types/port.go

```diff
@@ -0,0 +1,78 @@
+package types
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	connectiontypes "github.com/cosmos/ibc-go/v2/modules/core/03-connection/types"
+	porttypes "github.com/cosmos/ibc-go/v2/modules/core/05-port/types"
+)
+
+const (
+	// ControllerPortFormat is the expected port identifier format to which controller chains must conform
+	// See (TODO: Link to spec when updated)
+	ControllerPortFormat = "<app-version>.<controller-conn-seq>.<host-conn-seq>.<owner>"
+)
+
+// GeneratePortID generates an interchain accounts controller port identifier for the provided owner
+// in the following format:
+//
+// 'ics-27-<connectionSequence>-<counterpartyConnectionSequence>-<owner-address>'
```

Comment:
```suggestion
// 'ics-27-1.<connectionSequence>.<counterpartyConnectionSequence>.<owner-address>'
```

Reply:
Will leave this one out as I want to remove it completely and link to the spec when merged. The format is also defined as a const at the top of file so will be fine to remove it and point to the spec